File: solutions/HJK9810/2026-09-10/medium/1289.py
def algorithm():
    cases = list(input().strip())
    size = len(cases)
    base = ['0'] * size
    count = 0

    if cases[0] == '1':
        base[0] = '1'
        count = 1

    for idx in range(1, size):
        if cases[idx] != base[idx - 1]:
            base[idx] = '0' if cases[idx] == '0' else '1'
            count += 1
        else:
            base[idx] = '0' if base[idx - 1] == '0' else '1'

    return count

T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    print(f"#{test_case} {algorithm()}")

File: solutions/HJK9810/2026-09-10/medium/1289.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;

class Solution {
    public static void main(String args[]) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(input.readLine());

        for(int test_case = 1; test_case <= T; test_case++) {
            char[] base = input.readLine().toCharArray();
            char[] bit = new char[base.length];
            Arrays.fill(bit, '0');
            int count = 0;

            if (base[0] == '1') {
                bit[0] = '1';
                count = 1;
            }

            for (int idx = 1; idx < base.length; idx++) {
                if (base[idx] != bit[idx - 1]) {
                    count += 1;
                }
                bit[idx] = base[idx];
            }

            System.out.printf("#%d %d\n", test_case, count);
        }
    }
}
